homeassistant/components/sensor/bitcoin.py

```diff
@@ -0,0 +1,232 @@
+    'minutes_between_blocks': ['Time between Blocks', 'min'],
+    'number_of_transactions': ['No. of Transactions', ''],
+    'hash_rate': ['Hash rate', 'PH/s'],
+    'timestamp': ['Timestamp', ''],
+    'mined_blocks': ['Minded Blocks', ''],
+    'blocks_size': ['Block size', ''],
+    'total_fees_btc': ['Total fees', 'BTC'],
+    'total_btc_sent': ['Total sent', 'BTC'],
+    'estimated_btc_sent': ['Estimated sent', 'BTC'],
+    'total_btc': ['Total', 'BTC'],
+    'total_blocks': ['Total Blocks', ''],
+    'next_retarget': ['Next retarget', ''],
+    'estimated_transaction_volume_usd': ['Est. Transaction volume', 'USD'],
+    'miners_revenue_btc': ['Miners revenue', 'BTC'],
+    'market_price_usd': ['Market price', 'USD']
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the Bitcoin sensor. """
+
+    try:
+        from blockchain.wallet import Wallet
+        from blockchain import exchangerates, exceptions
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import blockchain. "
+            "Did you maybe not install the 'blockchain' package?")
+
+        return None
+
+    wallet_id = config.get('wallet', None)
+    password = config.get('password', None)
+    currency = config.get('currency', 'USD')
+
+    if currency not in exchangerates.get_ticker():
+        _LOGGER.error('Currency "%s" is not available. Using "USD".', currency)
+        currency = 'USD'
+
+    wallet = Wallet(wallet_id, password)
+
+    try:
+        wallet.get_balance()
+    except exceptions.APIException as e:
+        _LOGGER.error(e)
+        wallet = None
+
+    dev = []
+    if wallet is not None and password:
+        dev.append(BitcoinSensor('wallet', currency, wallet))
+
+    for variable in config['display_options']:
+        if variable['type'] not in OPTION_TYPES:
+            _LOGGER.error('Option type: "%s" does not exist', variable['type'])
+        else:
+            dev.append(BitcoinSensor(variable['type'], currency))
+
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class BitcoinSensor(Entity):
+    """ Implements a Bitcoin sensor. """
+
+    def __init__(self, option_type, currency, wallet=''):
+
+        self._name = OPTION_TYPES[option_type][0]
+        self._unit_of_measurement = OPTION_TYPES[option_type][1]
+        self._currency = currency
+        self._wallet = wallet
+        self.type = option_type
+        self._state = None
+        self.update()
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        return self._unit_of_measurement
+
+    # pylint: disable=too-many-branches
+    def update(self):
+        """ Gets the latest data and updates the states. """
+
+        stats = statistics.get()
```

Comment:
If you have three Bitcoin sensors, this will be called three times while only once is necessary for the latest updates. It would be better to only call it once every update cycle. You can use the `homeassistant.util.Throttle` decorator for this.